tests/setup_cfg_fmt_test.py

```diff
@@ -296,6 +296,60 @@ def test_adds_long_description_with_readme(filename, content_type, tmpdir):
     )
 
 
+@pytest.mark.parametrize(
+    'filename',
+    (
+        'README.rst',
+        'README.markdown',
+        'README.md',
+        'README',
+        'README.txt',
+        'readme.txt',
+    ),
+)
+def test_readme_discover_prefers_anything_over_asciidoc(filename, tmpdir):
+    tmpdir.join(filename).write('my project!')
+    other_files = [
+        'README.adoc',
+        'README.asciidoc',
+    ]
+    for other_file in other_files:
+        tmpdir.join(other_file).write('my project!')
+
+    setup_cfg = tmpdir.join('setup.cfg')
+    setup_cfg.write(
+        '[metadata]\n'
+        'name = pkg\n'
+        'version = 1.0\n',
+    )
+
+    assert main((str(setup_cfg),))
+
+    assert filename in setup_cfg.read()
+
+
+@pytest.mark.parametrize(
+    ('filename'),
+    (
+        ('README.adoc'),
+        ('README.asciidoc'),
+    ),
+)
+def test_readme_discover_uses_asciidoc_if_none_other_found(filename, tmpdir):
+    tmpdir.join(filename).write('my project!')
+
+    setup_cfg = tmpdir.join('setup.cfg')
+    setup_cfg.write(
+        '[metadata]\n'
+        'name = pkg\n'
+        'version = 1.0\n',
+    )
+
+    assert main((str(setup_cfg),))
+
+    assert filename in setup_cfg.read()
```

Comment:
this test is imprecise -- assert the actual output

Reply:
yes you're right, name and version will always stay the order they are in this universe so its actually better to do it like in the other tests with a precise output check :facepalm: